pulseapi/profiles/views.py

```diff
@@ -66,3 +67,50 @@ def put(self, request, *args, **kwargs):
             pass
 
         return super(UserProfileAPIView, self).put(request, *args, **kwargs)
+
+
+class ProfileCustomFilter(filters.FilterSet):
+    """
+      We add custom filtering to allow you to filter by:
+
+      * Profile type - pass the `?profile_type=` query parameter
+      * Program type - pass the `?program_type=` query parameter
+      * Program year - pass the `?program_year=` query parameter
+    """
+    profile_type = django_filters.CharFilter(
+        name='profile_type__value',
+        lookup_expr='iexact',
+    )
+    program_type = django_filters.CharFilter(
+        name='program_type__value',
+        lookup_expr='iexact',
+    )
+    program_year = django_filters.CharFilter(
+        name='program_year__value',
+        lookup_expr='iexact',
+    )
+
+    class Meta:
+        """
+        Required Meta class
+        """
+        model = UserProfile
+        fields = [
+            'profile_type',
+            'program_type',
+            'program_year',
+        ]
+
+
+class UserProfileAPISearchView(ListCreateAPIView):
+    serializer_class = UserProfileSerializer
```

Comment:
I don't know if this is the right serializer to use based on the views above. It looks like this serializer is used on permission protected views and for public views we use `UserProfilePublicSerializer`

Reply:
Hmm, I didn't see any results "not showing up" when calling the API from a non-authenticated session/browser, though.

Reply:
Interestingly, or serializer does not seem to say who the profile belongs to... if I set a custom profile name, that shows, but if I leave it empty, there is literally nothing to tie it to "a human"...

![image](https://user-images.githubusercontent.com/177243/35710076-604e6acc-076a-11e8-85f0-9fe76f8e1191.png)

probably needs a fix to the Profile serializer in general?

Reply:
Wait a second - why are those fields showing at all, `enable_extended_profile` is false...

Reply:
Okay so this is that weird situation: calling the route through DRF with `/api/pulse/profiles/search?profile_type=plain` for instance, the serializer receives `<QuerySet [<UserProfile: profile for admin@example.com>]>`, rather than that profile.